### DISCUSSION TITLE
Ability to choose a provider when using `middleware.js`

### DISCUSSION THREAD

**Discussion by erikash** [nextauthjs/next-auth#7488] [Help]
### Question 💬

How can I choose the provider I'm interested in when using the basic middleware to secure all pages of my site?

### How to reproduce ☕️

Add `middleware.ts` to the src folder
```
middleware.ts
export { default } from 'next-auth/middleware';
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by renet** [comment]
In other words, what you want to do is set-up the middleware in a way that it behaves just like `signIn("providerName")` does: Redirecting directly to the authentication provider's login page, instead of to the interstitial page (`/api/auth/signin`) where a user has to select a provider first, right?

If that's the case, that's exactly what I want, as well.

A general (but not so elegant) approach to get this would be a different feature: Automatically redirecting to the providers login page, whenever a user calls `/api/auth/signin` and there's only one provider available/configured.

Is any of those already possible somehow, or should I open a feature request?
